fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractCumulativeLoanScheduleGenerator.java

```diff
@@ -423,6 +427,87 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
                 scheduleParams.getTotalRepaymentExpected().getAmount(), totalOutstanding);
     }
 
+    private void addChargesOnlyRepaymentSchedule(MonetaryCurrency currency, LoanScheduleParams scheduleParams,
+            List<LoanScheduleModelPeriod> periods, Set<LoanCharge> loanCharges, MathContext mc, LoanApplicationTerms loanApplicationTerms,
+            Collection<RecalculationDetail> transactions,
+            LoanRepaymentScheduleTransactionProcessor loanRepaymentScheduleTransactionProcessor, final HolidayDetailDTO holidayDetailDTO,
+            final LocalDate lastRestDate, final LocalDate firstRepaymentDate, final LocalDate idealDisbursementDate) {
+        if (periods.size() > 0) {
+            LoanScheduleModelPeriod lastInstallment = periods.get(periods.size() - 1);
+            // get end date for the schedule from loan charges
+            LocalDate scheduleDueDate = null;
+            if (!(BigDecimal.ZERO.compareTo(loanApplicationTerms.getAnnualNominalInterestRate()) < 0)) {
+                for (final LoanCharge loanCharge : loanCharges) {
+                    if (loanCharge.isSpecifiedDueDate() && DateUtils.isAfter(loanCharge.getDueDate(), lastInstallment.periodDueDate())) {
+                        scheduleDueDate = loanCharge.getDueDate();
+                    }
+                }
+            }
+            if (scheduleDueDate != null) {
+                final Collection<LoanTermVariationsData> interestRates = loanApplicationTerms.getLoanTermVariations()
+                        .getInterestRateChanges();
+
+                LocalDate periodStartDateApplicableForInterest = calculateInterestStartDateForPeriod(loanApplicationTerms,
```

Comment:
I dont think we need to calculate interest after maturity date. @bharathc27 Thoughts?

Reply:
@bharathc27 Please comment.

Reply:
@ruchiD As discussed there is not interest calculcation involved for N+1 Installment that is after the maturity date.

N+1 Installment scenario happens only for 2 cases
1. for charges - here user entered amount will be the installment amount
2. for Charge-backs- here amounts are inherited from the repayment that charge-back was triggered from.

@adamsaghy cc